src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -1458,6 +1458,14 @@ public override IOperation VisitBinaryOperator(IBinaryOperation operation, int?
                                                 operation.OperatorMethod, semanticModel: null, operation.Syntax, operation.Type, operation.ConstantValue, IsImplicit(operation));
         }
 
+        public override IOperation VisitTupleBinaryOperator(ITupleBinaryOperation operation, int? captureIdForResult)
+        {
+            _evalStack.Push(Visit(operation.LeftOperand));
+            IOperation rightOperand = Visit(operation.RightOperand);
+            return new TupleBinaryOperatorExpression(operation.OperatorKind, _evalStack.Pop(), rightOperand,
+                semanticModel: null, operation.Syntax, operation.Type, operation.ConstantValue, IsImplicit(operation));
+        }
```

Comment:
The ordering for `(a, b) == (c, d)` is `tempA = a; tempB = b; tempC = c; tempD = d; tempA == tempC && tempB == tempD`.
That last expression has a mini control-flow. This can skip some conversions and also some user-defined comparison operators. 
Do we expect to represent that in the CFG?

Reply:
Based on our discussion, filed https://github.com/dotnet/roslyn/issues/26969 as a follow-up issue.

Reply:
Thanks for filing the issue. Definitely feels like something the CFG needs to express. There is no way we can expect consumers to be able to correctly infer the control flow + side effects here. Too complex. 

---
In reply to: [189291960](https://github.com/dotnet/roslyn/pull/26919#discussion_r189291960) [](ancestors = 189291960)